src/main/java/net/sf/jabref/gui/entryeditor/EntryEditor.java

```diff
@@ -288,32 +289,70 @@ private void setupFieldPanels() {
         }
 
         // other fields
-        List<String> displayedFields = Stream.concat(requiredFields.stream(), displayedOptionalFields.stream()).map(String::toLowerCase).collect(Collectors.toList());
-        List<String> otherFields = entry.getFieldNames().stream().map(String::toLowerCase).filter(f -> !displayedFields.contains(f)).collect(Collectors.toList());
+        // other fields also do not contain hidden fields
+        Set<String> displayedFields = Stream
+                .concat(type.getAllFields().stream(), Globals.prefs.getCustomTabFieldNames().stream())
+                .map(String::toLowerCase).flatMap(f -> Stream.of(f, "_" + f)).collect(Collectors.toSet());
+        List<String> otherFields = entry.getFieldNames().stream().map(String::toLowerCase)
+                .filter(f -> !displayedFields.contains(f)).collect(Collectors.toList());
         otherFields.remove(BibEntry.KEY_FIELD);
         otherFields.removeAll(Globals.prefs.getCustomTabFieldNames());
 
         if (!otherFields.isEmpty()) {
+
             addOtherTab(otherFields);
         }
 
+        addGeneralTab();
         // general fields from preferences
-        addGeneralTabs();
+        addAbstractReviewTabs();
         // source tab
         addSourceTab();
     }
 
-    private void addGeneralTabs() {
+    private void addAbstractReviewTabs() {
         EntryEditorTabList tabList = Globals.prefs.getEntryEditorTabList();
-        for (int i = 0; i < tabList.getTabCount(); i++) {
-            EntryEditorTab newTab = new EntryEditorTab(frame, panel, tabList.getTabFields(i), this, false,
-                    false, tabList.getTabName(i));
+        //i starts with 1 because 0 is the GeneralTab which has its own method
+        for (int i = 1; i < tabList.getTabCount(); i++) {
+            List<String> tabFields = new ArrayList<>();
```

Comment:
I think you could extract the for loop below into a new method ``public static ArrayList<String > addFieldAndHiddenField(ArrayList<String> list, String fieldName) {...}`` since you do this 3 times in this class.